### ISSUE TITLE
Reuse last resource from url dispatch table

### ISSUE THREAD

**Issue by asvetlov** [aio-libs/aiohttp#2586]
If name and path is the same -- add a route to existing resource instead of creating a new one.

E.g.
```
router.add_get('/', ...)
router.add_post('/', ...)
```
should create the single resource if path and name are the same.

It reduces resources table size and boosts url lookup if handlers for the same url is grouped together.

The change is easy, Pull Request is welcome.

**Comment by sheb** [comment]
I'm working on it

**Comment by sheb** [comment]
Hmm it's not allowed to add a route with existing path and name as it's already handled by 

> aiohttp.web_urldispatcher.UrlDispatcher().register_resource(..)

See below
```
app.router.add_get("/", lambda x: x, name="foo")
app.router.add_post("/", lambda x: x, name="foo")
>>> ValueError: Duplicate 'foo', already handled by <PlainResource 'foo'  /
```

Should I match only the path ?

**Comment by asvetlov** [comment]
The feature should check both `path` and `name` for last added resource.
If they are matched to added one -- return last item.
All functionality could be done in `.add_resource()` method.

**Comment by sheb** [comment]
Ok got it, pull request soon

**Comment by asvetlov** [comment]
Fixed by #2603
